src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/BaseTest.cs

```diff
@@ -439,8 +439,13 @@ protected virtual void CleanupTest ()
 				return;
 			if (TestContext.CurrentContext.Result.Outcome.Status == NUnit.Framework.Interfaces.TestStatus.Passed || 
 			    TestContext.CurrentContext.Result.Outcome.Status == NUnit.Framework.Interfaces.TestStatus.Skipped) {
-				FileSystemUtils.SetDirectoryWriteable (output);
-				Directory.Delete (output, recursive: true);
+				try {
+					FileSystemUtils.SetDirectoryWriteable (output);
+					Directory.Delete (output, recursive: true);
+				} catch (Exception ex) {
+					// Don't fail the test if this occurs, but log instead
+					TestContext.Out.WriteLine ($"[TESTLOG] Test {TestName} failed to delete `{output}`: {ex}");
```

Comment:
After reviewing the build artifacts with this in place, it made me wonder @dellis1972 if something is going on with aapt2: https://github.com/xamarin/xamarin-android/commit/0d6fd83b3299406225015a4e381f252428926413

I'm seeing a lot of test directories that couldn't be deleted, and they have a `obj\Debug\lp\res` directory. Is it possible we aren't waiting long enough for `aapt2` to finish?

Reply:
Are the processes still running? I seem to remember they weren't last time we checked on this.
Also aapt2 daemons have a working directory in the temp folder, so if they specifically won't lock any project directory even if they are running but not doing anything. 

So that leaves the possibility that they are still running and doing work, in which case why is the test over? very confusing.

Reply:
`aapt2` isn't running at the end of the test run.

Is it possible it is not really finished at the point we escape this loop:

https://github.com/xamarin/xamarin-android/blob/ca015b1147f51ae1066cc12d36efc9e8df56299b/src/Xamarin.Android.Build.Tasks/Utilities/Aapt2Daemon.cs#L217-L219

Are there some cases where it's not running this loop at all?